franka_gazebo/src/franka_hw_sim.cpp

```diff
@@ -480,11 +558,9 @@ void FrankaHWSim::updateRobotState(ros::Time time) {
     this->robot_state_.tau_J[i] = joint->effort;
     this->robot_state_.dtau_J[i] = joint->jerk;
 
-    // since we don't support position or velocity interface yet, we set the desired joint
-    // trajectory to zero indicating we are in torque control mode
-    this->robot_state_.q_d[i] = joint->position;  // special case for resetting motion generators
-    this->robot_state_.dq_d[i] = 0;
-    this->robot_state_.ddq_d[i] = 0;
+    this->robot_state_.q_d[i] = joint->position;
```

Comment:
@gollth Since there is a lot of interest in this pull request, I again rebased it on the `develop` branch. I also fixed the conflicts that were introduced due to https://github.com/frankaemika/franka_ros/pull/198 being merged. Here I undid some changes made in  https://github.com/frankaemika/franka_ros/pull/198 since these modes are also supported after #181 is merged. You can of course, also add some conditional logic using the `joint_control_methods_[joint->name]` variable. I will leave this to you.